From e3a4cb15063ab8ab6839fecee805a42b27914a2a Mon Sep 17 00:00:00 2001
From: Martin Wilck <mwilck@suse.com>
Date: Tue, 9 Apr 2024 15:09:59 +0200
Subject: [PATCH 179/179] todaysFactsWidget: Fix broken layout in GNOME 45 and
 46

It turns out that cc353e2 worked on GNOME 46 for me, but not
for everyone. So use the GNOME shell version check instead.
---
 extension/widgets/todaysFactsWidget.js | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/extension/widgets/todaysFactsWidget.js b/extension/widgets/todaysFactsWidget.js
index fe35ba2..29b9bbd 100644
--- a/extension/widgets/todaysFactsWidget.js
+++ b/extension/widgets/todaysFactsWidget.js
@@ -28,6 +28,7 @@ import GObject from 'gi://GObject';
 
 import { gettext as _ } from 'resource:///org/gnome/shell/extensions/extension.js';
 import * as Stuff from '../stuff.js';
+import * as Config from 'resource:///org/gnome/shell/misc/config.js';
 
 /**
  * A widget that lists all facts for *today*.
@@ -48,7 +49,10 @@ class TodaysFactsWidget extends St.ScrollView {
             reactive: true
         });
         this.factsBox.add_child(this.facts_widget);
-        this.add_actor(this.factsBox);
+        if (Config.PACKAGE_VERSION.substring(0, 2) == "45")
+            this.add_actor(this.factsBox);
+        else
+            this.add_child(this.factsBox);
 
     }
 
-- 
2.44.0

